django/contrib/admin/tests.py

```diff
@@ -97,14 +99,29 @@ def wait_until_invisible(self, css_selector, timeout=10):
             timeout
         )
 
-    def wait_page_loaded(self):
+    def wait_page_ready(self, timeout=10):
+        """
+        Block until page has started to load.
+        """
+        self.wait_until(
+            lambda driver: driver.execute_script('return document.readyState;') == 'complete',
+            timeout
+        )
+
+    @contextmanager
+    def wait_page_loaded(self, timeout=10):
         """
         Block until page has started to load.
         """
         from selenium.common.exceptions import TimeoutException
+        from selenium.webdriver.support import expected_conditions as ec
+
         try:
+            old_page = self.selenium.find_element_by_tag_name('html')
+            yield
             # Wait for the next page to be loaded
-            self.wait_for('body')
+            self.wait_until(ec.staleness_of(old_page), timeout=timeout)
+            self.wait_page_ready(timeout=timeout)
         except TimeoutException:
             # IE7 occasionally returns an error "Internet Explorer cannot
```

Comment:
Do we even need this anymore? IE7 is *long* gone.

It also swallows all the `TimeoutException`s that we're expecting to happen here with the use of `wait_until()`, etc? Wouldn't we rather know that it timed out and we knew that was the cause than falling through to an obscure failure in a later assertion?

Reply:
I don't know, but good questions: @carltongibson do we test on Jenkins against IE7?

Reply:
I removed the try...except... — Thanks both.